tests/integration/test_types_integration.py

```diff
@@ -0,0 +1,278 @@
+        .add_field(sql="cast(-9223372036854775808 AS BIGINT)", python=-9223372036854775808) \
+        .add_field(sql="cast(null AS BIGINT)", python=None) \
+        .execute()
+
+
+def test_real_types(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="cast(3.4028235E38 AS REAL)", python=3.4028235e+38) \
+        .add_field(sql="cast(1.4E-45 AS REAL)", python=1.4e-45) \
+        .add_field(sql="cast('Infinity' AS REAL)", python=math.inf) \
+        .add_field(sql="cast('-Infinity' AS REAL)", python=-math.inf) \
+        .add_field(sql="cast('NaN' AS REAL)", python=math.nan) \
+        .add_field(sql="cast(null AS REAL)", python=None) \
+        .execute()
+
+
+def test_double_types(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="cast(1.7976931348623157E308 AS DOUBLE)", python=1.7976931348623157e+308) \
+        .add_field(sql="cast(4.9E-324 AS DOUBLE)", python=5e-324) \
+        .add_field(sql="cast('Infinity' AS DOUBLE)", python=math.inf) \
+        .add_field(sql="cast('-Infinity' AS DOUBLE)", python=-math.inf) \
+        .add_field(sql="cast('NaN' AS DOUBLE)", python=math.nan) \
+        .add_field(sql="cast(null AS DOUBLE)", python=None) \
+        .execute()
+
+
+def test_decimal_types(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="DECIMAL '10.3'", python=Decimal('10.3')) \
+        .add_field(sql="cast(null AS DECIMAL)", python=None) \
+        .add_field(sql="cast('0.123456789123456789' AS DECIMAL(18,18))", python=Decimal('0.123456789123456789')) \
+        .add_field(sql="cast(null AS DECIMAL(18,18))", python=None) \
+        .add_field(sql="cast('10.3' AS DECIMAL(38,1))", python=Decimal('10.3')) \
+        .add_field(sql="cast('0.123456789123456789' AS DECIMAL(18,2))", python=Decimal('0.12')) \
+        .add_field(sql="cast(null AS DECIMAL(38,0))", python=None) \
+        .execute()
+
+
+def test_string_types(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="'aaa'", python='aaa') \
+        .add_field(sql="U&'Hello winter \2603 !'", python='Hello winter °3 !') \
+        .add_field(sql="cast(null AS VARCHAR)", python=None) \
+        .add_field(sql="cast('bbb' AS VARCHAR(1))", python='b') \
+        .add_field(sql="cast(null AS VARCHAR(1))", python=None) \
+        .add_field(sql="cast('ccc' AS CHAR)", python='c') \
+        .add_field(sql="cast(null AS CHAR)", python=None) \
+        .add_field(sql="cast('ddd' AS CHAR(1))", python='d') \
+        .add_field(sql="cast('😂' AS CHAR(1))", python='😂') \
+        .add_field(sql="cast(null AS CHAR(1))", python=None) \
+        .add_field(sql="X'65683F'", python='ZWg/') \
+        .add_field(sql="cast(null AS VARBINARY)", python=None) \
+        .add_field(sql="cast('{}' AS JSON)", python='"{}"') \
+        .add_field(sql="cast('null' AS JSON)", python='"null"') \
+        .add_field(sql="cast(null AS JSON)", python=None) \
```

Comment:
varbinary and JSON into their own methods.

Also let's test some varbinary values which don't represent valid strings.

Reply:
Also empty varbinary (not null) is special as well.

Reply:
for JSON we should also test float values, strings, strings with special chars (", ', :), empty array as well.

Reply:
Actually right now we don't seem to perform anything special treatment for JSON

Reply:
Done for splitting each type in its own test

Reply:
Either add TODO to add the missing tests or add the tests for the other values and assert that they fail.

Reply:
What test are you talking about here? JSON tests (even though we just convert them to strings), varbinary which represent invalid strings (e.g. non-UTF-8?) or something else?

Reply:
Both.

- Add test for varbinary which doesn't represent a string
- Add test for JSON with:
  - JSON '3.14'
  - JSON 'a string'
  - JSON 'a " complex \' string :' (I'm not sure if I escaped that `'` correctly)
  - JSON '[]'

Even if the test fails we should assert on that failure  so that when something like https://github.com/trinodb/trino-python-client/pull/214 changes behaviour we know to update the tests. It also serves as documentation of the "issues" we need to fix.

Reply:
I'm not entirely sure I know exactly what you have in mind but I've added such tests.